### PR TITLE
Created Book Detail

### PR DESCRIPTION
**Pull request by eldani17**
## Summary
- The home screen is created to test the components.
- The Button component is created, to which the desired style can be passed as props.
- The ItemDetail component is created.

#### Galaxy S (or similar)
image

#### Pixel 3XL (or similar)
image

## Trello Card
https://trello.com/c/hGTHdKCD/17-n-72-detalle-del-libro

### CODE REVIEW

File: wbooks/src/app/components/button/index.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  stylesButton: string;
+}
+
+function Button({ text, stylesButton }: Props) {
+  const { styleText, styleBackground } = getButtonStyle(stylesButton);
+  return (
+    <TouchableOpacity>
+      <View style={[styles.btnDefault, styleBackground]}>
+        <Text style={[styles.btnTextDefault, styleText]}> {text} </Text>
+      </View>
```
Comment (mattgle) [inline wbooks/src/app/components/button/index.tsx]: the view is unnecessary u can send the styles to the touchableOpacity.

File: wbooks/src/app/components/button/styles.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { StyleSheet } from 'react-native';
+import { cerulean, transparent, white } from '@constants/colors';
+
+export default StyleSheet.create({
+  btnDefault: {
+    borderRadius: 50,
+    paddingVertical: 10,
+    paddingHorizontal: 10,
```
Comment (mattgle) [inline wbooks/src/app/components/button/styles.ts]: change this to padding: 10

File: wbooks/src/app/components/button/utils.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+export enum BUTTONTHEME {
+  OUTLINE = 'Outline',
+  GRADIENTS = 'Gradients'
+}
+
+export interface Props {
+  styleText?: Object;
+  styleBackground?: Object;
+}
```
Comment (mattgle) [inline wbooks/src/app/components/button/utils.ts]: change name

File: wbooks/src/app/components/button/utils.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+export function getButtonStyle(type: string) {
```
Comment (mattgle) [inline wbooks/src/app/components/button/utils.ts]: type should be BUTTONNTHEME not string

File: wbooks/src/app/components/button/utils.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  const result: Props = {};
```
Comment (mattgle) [inline wbooks/src/app/components/button/utils.ts]: delete

File: wbooks/src/app/components/button/utils.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+      result.styleText = styles.btnTextGradients;
+      result.styleBackground = styles.btnGradients;
+      break;
+    }
+    default: {
+      break;
+    }
+  }
+  return result;
+}
```
Comment (mattgle) [inline wbooks/src/app/components/button/utils.ts]:
```suggestion
  switch (type) {
    case BUTTONTHEME.OUTLINE: {
    return {
    styleText:styles.btnTextOutline,
    styleBackground: styles.btnOutline
}
    }
    case BUTTONTHEME.GRADIENTS: {
    return {
     styleText: styles.btnTextGradients;
     styleBackground: styles.btnGradients;
}
    }
    default: {
      return {};
    }
  }
  return result;
}
```

File: wbooks/src/app/components/itemDetail/index.tsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { Text, View, Image } from 'react-native';
+import Button from '@components/button';
```
Comment (mattgle) [inline wbooks/src/app/components/itemDetail/index.tsx]: Change name of the component to CustomButton

File: wbooks/src/app/components/itemDetail/styles.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    maxWidth: 150
+  },
+  author: {
+    color: dustyGray,
+    fontSize: 16
+  },
+  year: {
+    color: dustyGray,
+    fontSize: 16
+  },
```
Comment (mattgle) [inline wbooks/src/app/components/itemDetail/styles.ts]: same, delete one.

File: wbooks/src/app/components/itemDetail/styles.ts [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+  genre: {
+    color: dustyGray,
+    fontSize: 16,
+    textTransform: 'capitalize'
+  }
```
Comment (mattgle) [inline wbooks/src/app/components/itemDetail/styles.ts]: use the preview style and add the textTransform.

**Review by mattgle — CHANGES_REQUESTED** [review]
great work! Some changes!

File: wbooks/src/app/components/button/index.tsx (hunk 1 above)
Comment (eldani17) [reply wbooks/src/app/components/button/index.tsx]: The View is removed and the styles are passed to the TouchableOpacity

File: wbooks/src/app/components/button/styles.ts (hunk 2 above)
Comment (eldani17) [reply wbooks/src/app/components/button/styles.ts]: only padding is unified

File: wbooks/src/app/components/button/utils.ts (hunk 4 above)
Comment (eldani17) [reply wbooks/src/app/components/button/utils.ts]: the data type is changed

File: wbooks/src/app/components/button/utils.ts (hunk 3 above)
Comment (eldani17) [reply wbooks/src/app/components/button/utils.ts]: the Props interface is removed

File: wbooks/src/app/components/button/utils.ts (hunk 5 above)
Comment (eldani17) [reply wbooks/src/app/components/button/utils.ts]: the const result is removed

File: wbooks/src/app/components/button/utils.ts (hunk 6 above)
Comment (eldani17) [reply wbooks/src/app/components/button/utils.ts]: switch is modified

File: wbooks/src/app/components/itemDetail/index.tsx (hunk 7 above)
Comment (eldani17) [reply wbooks/src/app/components/itemDetail/index.tsx]: the name of the Button component is changed to CustomButton

File: wbooks/src/app/components/itemDetail/styles.ts (hunk 8 above)
Comment (eldani17) [reply wbooks/src/app/components/itemDetail/styles.ts]: the year style is removed and the author name is changed to subTitle

File: wbooks/src/app/components/itemDetail/styles.ts (hunk 9 above)
Comment (eldani17) [reply wbooks/src/app/components/itemDetail/styles.ts]: the color and fontSize of the genre style is eliminated

File: wbooks/src/app/components/customButton/utils.ts [hunk 10]
```diff
@@ -0,0 +1,20 @@
+import styles from './styles';
+
+export enum BUTTONTHEME {
```
Comment (mattgle) [inline wbooks/src/app/components/customButton/utils.ts]:
```suggestion
export enum CustomButtonThemes {
```

**Review by mattgle — DISMISSED** [review]
Good job! Just this little change

**Review by mattgle — APPROVED** [review]

**Pull request merged**